tools/structure/src/main.py

```diff
@@ -47,21 +48,25 @@ def run(
 
         self.stream_log("Indexing document...")
         file_hash = self.get_exec_metadata.get(MetadataKey.SOURCE_HASH)
-        tool_index = ToolIndex(tool=self)
+        index = Index(tool=self)
         tool_id = tool_metadata[SettingsKeys.TOOL_ID]
         outputs = tool_metadata[SettingsKeys.OUTPUTS]
         try:
+            usage_kwargs: dict[Any, Any] = dict()
+            run_id = CommonUtils.get_uuid()
+            usage_kwargs["run_id"] = run_id
```

Comment:
@gaya3-zipstack since the structure tool runs in a workflow / pipeline - it will have the `workflow_id` and `execution_id` passed to it through the ToolMetadata and [available in self](https://github.com/Zipstack/unstract/blob/3b787e87adde430e802e2b007f27a4d831dfa430/tools/classifier/src/main.py#L89-L90). It'll help us eventually correlate the token usage to workflows maybe

Reply:
As runId discussions are happening now, will look into this from that context also later @chandrasekharan-zipstack  I am still unclear on how the schema will look like for ETL/Workflows. Basically wherever tools are going to be involved....

Reply:
Let's carry that discussion separately today along with others